src/App.tsx

```diff
@@ -1,14 +1,45 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [todoId, setTodoId] = useState(0);
+  const [filter, setFilter] = useState('all');
+  const [responseFilter, setResponseFilter] = useState('');
+
+  useEffect(() => {
+    getTodos()
+      .then((response) => {
+        setTodos(response);
+        setLoading(false);
+      });
```

Comment:
Let’s use async await. It gives you a simpler way to work with asynchronous promise-based code